nebullvm/inference_learners/tensor_rt.py

```diff
@@ -248,6 +247,59 @@ def load(cls, path: Union[Path, str], **kwargs):
         )
 
 
+class PytorchTensorRTInferenceLearner(PytorchBaseInferenceLearner):
+    MODEL_NAME = "model_optimized.pt"
+
+    def __init__(
+        self, torch_model: torch.jit.ScriptModule, dtype: torch.dtype, **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.model = torch_model.eval()
+        if torch.cuda.is_available():
+            self.model.cuda()
+        self.dtype = dtype
+
+    def run(self, *input_tensors: torch.Tensor) -> Tuple[torch.Tensor, ...]:
+        device = input_tensors[0].device
+        if torch.cuda.is_available():
+            if self.dtype == torch.half:
+                input_tensors = (t.cuda().half() for t in input_tensors)
```

Comment:
Note that model.half() does not change the embedding parameters (for instance in the first transformer layer), so casting the input to half can give an error when using half precision with transformers. Just float32 inputs should be converted to half precision.

Reply:
Note that this should also be reflected in the HalfPrecisionTransformation (in nebullvm.transformations.precision_tfms).